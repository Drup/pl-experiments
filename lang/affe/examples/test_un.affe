let f = fun x -> fun y -> x
let a x = f 2 x
let a2 = a 3
let b = fun x -> f x x
