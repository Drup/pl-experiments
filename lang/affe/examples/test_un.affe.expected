f :
  ∀^k, ^l, 'a, 'b. ('b : ^l) & ('a : ^k) & (^l < affₙ) =>
  'a -> 'b -{^k}> 'a

a : ∀^k, 'a. ('a : ^k) & (^k < affₙ) => 'a -> int

a2 : int

b : ∀^k, 'a. ('a : ^k) & (^k < unₙ) => 'a -> 'a

